core/services/functions/listener.go

```diff
@@ -352,7 +366,31 @@ func (l *FunctionsListener) handleRequest(ctx context.Context, requestID [32]byt
 		return
 	}
 
-	computationResult, computationError, domains, err := eaClient.RunComputation(ctx, requestIDStr, l.job.Name.ValueOrZero(), subscriptionOwner.Hex(), subscriptionId, "", requestData)
+	nodeProvidedSecrets := ""
+	if l.decryptor != nil && requestData.SecretsLocation == LocationRemote && len(requestData.Secrets) > 0 {
+		thresholdEncSecrets, userError, err2 := eaClient.FetchEncryptedSecrets(ctx, requestData.Secrets, requestIDStr, l.job.Name.ValueOrZero())
+
+		// To avoid a breaking change, if secrets fetching is unsuccessful,
+		// proceed by allowing the adapter handle secrets as before.
+		// Eventually, this will be deprecated and the error will be returned to the user on-chain
+		// by uncommenting the lines within the following if statements.
+		if err2 != nil {
+			l.logger.Errorw("failed to fetch encrypted secrets", "requestID", requestIDStr, "err", err2)
+		}
+		if len(userError) != 0 {
+			l.logger.Debugw("user error while fetching threshold encrypted secrets - skipping threshold decryption", "requestID", requestIDStr, "err", string(userError))
```

Comment:
Maybe avoid "error" in the message. How about "no threshold decryption detected, falling back to legacy secrets" or something along those lines?